community/kernel/src/main/java/org/neo4j/kernel/api/security/AuthSubject.java

```diff
@@ -54,6 +54,22 @@
     String username();
 
     /**
+     * @param username a username
+     * @return true if the provided username is the underlying user name of this subject
+     */
+    boolean hasUsername( String username );
+
+    /**
+     * Ensure that the provided username is the name of an existing user known to the system.
+     *
+     * @param username a username
+     * @throws InvalidArgumentsException if the provided user name is not the name of an existing user
+     */
+    default void ensureUserExistsWithName( String username ) throws InvalidArgumentsException {
```

Comment:
I think factoring this out just complicates things uneccesarliy and having the default implementation generate the correct errors seems reasonable to me so I'm not going to touch it.